src/library_fs.js

```diff
@@ -1881,7 +1877,8 @@ FS.staticInit();` +
           if (onload) onload();
           removeRunDependency(dep);
         }
-        if (Browser.handledByPreloadPlugin(byteArray, fullname, finish, () => {
+        if (typeof Browser != 'undefined' &&
```

Comment:
Can Browser ever actually be undefined here though?  if so how did this code work previously?

Reply:
Hmm, good question. It looks like in practice if you set `emcc --use-preload-plugins` then we auto-include `$Browser` in `emcc.py`. Maybe that's enough of a signal, then, to include this code only in that case. I'll try that.

Reply:
Ok, rewritten now. A little more code, but it is more precise now about when we need preload plugins in an explicit way.

That is, how this worked before is that Browser happened to be present in all the paths that used preload plugins, either it was added manually or other APIs pulled it in. With this PR, we have a new internal setting for when we need preload plugins and it is set in the right places (I hope).